.github/workflows/pr_test.yml

```
@@ -0,0 +1,76 @@
+on: pull_request
+name: Test
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v1
+    - name: Install Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.14.x
+    - name: Set GOPATH
+      # Temporary fix, see: https://github.com/actions/setup-go/issues/14
+      run: |
+        echo "::set-env name=GOPATH::$(go env GOPATH)"
+        echo "::add-path::$(go env GOPATH)/bin"
+      shell: bash
+    - name: Restore cache
+      id: cache
+      uses: actions/cache@v1
+      with:
+        path: ~/go/bin
+        key: ${{ runner.os }}-go-bin-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-bin-
+    - name: Restore cache
+      uses: actions/cache@v1
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-mod-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-mod-
+    - name: Check Module
+      run: make check-mod
+      shell: bash
+    - name: Check Imports
+      run: make check-imports
+      shell: bash
+    - name: Check Fmt
+      run: make check-fmt
+      shell: bash
+    - name: Vet
+      run: make vet
+      shell: bash
+    - name: Staticcheck
+      run: make staticcheck
+      shell: bash
+  test:
+    strategy:
+      matrix:
+        go-version: [1.14.x]
+        platform: [ubuntu-latest, macos-latest, windows-latest]
```

I'm personally happy for us to only test this project on a linux architecture (which is what we currently do in Travis), we use all three in the CLI project as we distribute a binary across all architectures. However as this is a package that gets consumed by others I don't think its necessary and slows down PR builds considerably. 
```suggestion
        platform: [ubuntu-latest]
```